### PR TITLE
Config docker image in CI

### PR DESCRIPTION
**Pull request by jakirkham**
Borrows some code from the feedstocks to configure the Docker image used either through environment variable or shyaml with a fallback to the default. Also moves the Docker image name to the CI config.

### CODE REVIEW

**Comment by jakirkham** [comment]
Should be ready for review @conda-forge/core 🙂

**Comment by jakirkham** [comment]
cc @conda-forge/staged-recipes

File: .circleci/run_docker_build.sh [hunk 1]
```diff
@@ -21,6 +20,16 @@ if hash docker-machine 2> /dev/null && docker-machine active > /dev/null; then
     HOST_USER_ID=$(docker-machine ssh $(docker-machine active) id -u)
 fi
 
+if [ -z "${IMAGE_NAME}" ]; then
+    SHYAML_INSTALLED="$(shyaml -h || echo NO)"
```
Comment (isuruf) [inline .circleci/run_docker_build.sh]: Do we install shyaml anywhere?

Comment (jakirkham) [reply .circleci/run_docker_build.sh]: Yeah it is already being installed in the CI configuration files.

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**